fuse/utils/data/collate.py

```diff
@@ -98,22 +94,15 @@ def _collect_values_to_list(self, samples: List[str], key: str) -> Tuple[List, b
         has_missing_values = False
         collected_values = []
         for index, sample in enumerate(samples):
-            sample = NDict(sample)
-            if key not in sample:
+            try:
+                value = sample[key]
+            except:
                 has_error = True
                 has_missing_values = True
                 if self._raise_error_key_missing:
                     raise Exception(f"Error: key {key} does not exist in sample {index}: {sample}")
                 else:
                     value = None
-            else:
```

Comment:
removing to optimize the running time.
Detecting NaNs and more could move to an optional op (in the data pipeline)